chaincode/input.go

```diff
@@ -90,6 +90,24 @@ type inputTraintuple struct {
 	Tag            string   `validate:"omitempty,lte=64" json:"tag"`
 }
 
+// inputCompositeTraintuple is the representation of input args to register a Traintuple
+type inputCompositeTraintuple struct {
+	AlgoKey        string              `validate:"required,len=64,hexadecimal" json:"algoKey"`
+	ObjectiveKey   string              `validate:"required,len=64,hexadecimal" json:"objectiveKey"`
+	InModelHead    inputCompositeModel `validate:"omitempty" json:"inModelHead"`
+	InModelTrunk   inputCompositeModel `validate:"omitempty" json:"inModelTrunk"`
+	DataManagerKey string              `validate:"required,len=64,hexadecimal" json:"dataManagerKey"`
+	DataSampleKeys []string            `validate:"required,unique,gt=0,dive,len=64,hexadecimal" json:"dataSampleKeys"`
+	ComputePlanID  string              `validate:"omitempty" json:"computePlanID"`
+	Rank           string              `validate:"omitempty" json:"rank"`
+	Tag            string              `validate:"omitempty,lte=64" json:"tag"`
+}
+
+type inputCompositeModel struct {
```

Comment:
On second thought, I think `inputModelComposite` would be better, what do you think @thibaultrobert ? IMO having the suffix `Composite` makes it clearer that we're talking about a `Composite`-specific case of an `InModel`.

Reply:
I'm gonna go with `Composite` suffix for the "out model" struct for now, I can update depending your response :)